### PR TITLE
Move root.Root from nevow to twisted.web.template

### PR DESCRIPTION
**Pull request by sajith**
Related to [3305](https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3305).  

`2to3 src/allmydata/web/root.py` and `pylint --py3k src/allmydata/web/root.py` seems to suggest a few more changes, but those should perhaps be a separate PR.

----

Here's a screenshot after this change:

root-after

----

Here's a screenshot fro before:

root-before

---
This change is [Reviewable](https://reviewable.io/reviews/tahoe-lafs/tahoe-lafs/710)

### CODE REVIEW

File: src/allmydata/test/web/test_root.py [hunk 1]
```diff
@@ -90,13 +83,28 @@ def test_missing(self):
...
-        res = r.render_service_row(ctx, s)
-        self.assertIdentical(res, ctx)
-        self.assertEqual(ctx.slots["version"], "")
-        self.assertEqual(ctx.slots["nickname"], "")
+        srv = NativeStorageServer("server_id", ann, None, {}, EMPTY_CLIENT_CONFIG)
+        srv.get_connection_status = lambda: ConnectionStatus(False, "summary", {}, 0, 0)
+
+        class FakeClient(_Client):
+            def __init__(self):
+                service.MultiService.__init__(self)
```
[the remark this answers is not in the document]
Comment (sajith) [reply src/allmydata/test/web/test_root.py:91]: I copied this from `test/.web/test_web.py`, which seems to to initialize a small enough client by avoiding the `super()` call.

https://github.com/tahoe-lafs/tahoe-lafs/blob/e2c6d288bf7ca9061ae7effc1228368d8f8d1bcd/src/allmydata/test/web/test_web.py#L256-L260

It seems that making a `super()` call would require a lot more machinery?

https://github.com/tahoe-lafs/tahoe-lafs/blob/e2c6d288bf7ca9061ae7effc1228368d8f8d1bcd/src/allmydata/client.py#L668-L669

File: src/allmydata/web/root.py [hunk 2]
```diff
@@ -320,113 +351,60 @@ def render_services(self, ctx, data):
             available = stats.get("storage_server.disk_avail")
             if available is not None:
                 msg += ", %s available" % abbreviate_size(available)
-            ul[T.li[T.a(href="storage")["Storage Server"], ": ", msg]]
+            ul(tags.li(tags.a("Storage Server", ": ", msg, href="storage")))
         except KeyError:
```
[the remark this answers is not in the document]
Comment (sajith) [reply src/allmydata/web/root.py:355]: Ah good catch, thank you!  I have not tested this well enough.  I will try again.

File: src/allmydata/web/root.py [hunk 3]
```diff
@@ -437,123 +415,147 @@ def data_connected_to_helper(self, ctx, data):
...
-        form = T.form(action="uri", method="get",
-                      enctype="multipart/form-data")[
-            T.fieldset[
-            T.legend(class_="freeform-form-label")["View a file or directory"],
-            "Tahoe-URI to view:"+SPACE,
-            T.input(type="text", name="uri"), SPACE*2,
-            T.input(type="submit", value="View!"),
-            ]]
-        return T.div[form]
-
```
[the remark this answers is not in the document]
Comment (sajith) [reply src/allmydata/web/root.py:500]: The forms are also in `web/welcome.xhtml`, so it seemed that equivalent renderers are redundant in `web/root.py`.  I removed them in cce6f83.

File: src/allmydata/test/web/test_root.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (meejah) [reply src/allmydata/test/web/test_root.py:91]: hrrrmmm, okay interesting. You should at least copy the comment (or something similar) too then :)

File: src/allmydata/web/root.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (meejah) [reply src/allmydata/web/root.py:500]: Ahh, okay, cool thanks.

File: src/allmydata/web/root.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sajith) [reply src/allmydata/web/root.py:355]: Changed it in 96c8c5e.  Below is a screenshot of how it is now:

Screenshot from 2020-05-05 09-04-59

And this is how it used to be before, in the nevow version:

Screenshot from 2020-05-05 09-03-59

**Pull request merged**